test/e2e/common/k8s.py

```diff
@@ -253,54 +311,71 @@ def wait_on_condition(reference: CustomResourceReference,
-        resource = get_resource(reference)
 
-        if 'conditions' not in resource['status']:
-            logging.error(f"Resource {reference} does not have a .status.conditions field.")
-            return False
-
-        desired_condition = None
-        for condition in resource['status']['conditions']:
-            if condition['type'] == condition_name:
-                desired_condition = condition
-
-        if not desired_condition:
-            logging.error(f"Resource {reference} does not have a condition of type {condition_name}.")
-            return False
-        else:
-            if desired_condition['status'] == desired_condition_status:
-                logging.info(f"Condition {condition_name} has status {desired_condition_status}, continuing...")
-                return True
+        desired_condition = get_resource_condition(reference, condition_name)
+        if desired_condition is not None and desired_condition['status'] == desired_condition_status:
+            logging.info(f"Condition {condition_name} has status {desired_condition_status}, continuing...")
+            return True
 
         sleep(period_length)
 
-    logging.error(f"Wait for condition {condition_name} to reach status {desired_condition_status} timed out")
+    if not desired_condition:
+        logging.error(f"Resource {reference} does not have a condition of type {condition_name}.")
+    else:
+        logging.error(f"Wait for condition {condition_name} to reach status {desired_condition_status} timed out")
     return False
 
-def is_resource_in_terminal_condition(
-        reference: CustomResourceReference, expected_substring: str):
+def get_resource_condition(reference: CustomResourceReference, condition_name: str):
+    """
+    Returns the required condition from .status.conditions
+
+    Precondition:
+        resource must exist in the cluster
+
+    Returns:
+        condition json if it exists. None otherwise
+    """
     if not get_resource_exists(reference):
         logging.error(f"Resource {reference} does not exist")
-        return False
+        return None
 
     resource = get_resource(reference)
-    terminal_status = _get_terminal_condition(resource)
-    # Ensure the status existed
-    if terminal_status is None:
-        logging.error(f"Expected .ACK.Terminal to exist in {reference}")
-        return False
+    if 'status' not in resource or 'conditions' not in resource['status']:
+        logging.error(f"Resource {reference} does not have a .status.conditions field.")
+        return None
 
-    if not terminal_status:
-        logging.error(
-            f"Expected terminal condition for resource {reference} to be true")
-        return False
+    for condition in resource['status']['conditions']:
+        if condition['type'] == condition_name:
+            return condition
+
+    return None
+
+def assert_condition_state_message(reference: CustomResourceReference,
```

Comment:
Not seeing where assert_condition_state_message is used

Reply:
test/e2e/sagemaker/tests/test_endpoint.py line #324